### PR TITLE
Scala 2.13.12 release notes draft

### PR DESCRIPTION
**Pull request by SethTisue**
[rendered](https://github.com/SethTisue/scala-dev/blob/scala-2.13.12/releases/2.13.12.md)

feedback encouraged

no suggestion too small

let's make this as clear and helpful as it can be

### CODE REVIEW

File: releases/2.13.12.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+The [release notes for Scala 2.13.0](https://github.com/scala/scala/releases/v2.13.0) have important information applicable to the whole 2.13 series.
+
+## Obtaining Scala
+
+Scala releases are available through a variety of channels, including (but not limited to):
+
+* Bump the `scalaVersion` setting in your sbt project
+* Bump the `using scala` setting in your Scala-CLI project
+* Download a distribution from [scala-lang.org](https://scala-lang.org/download/2.13.11.html)
+* Obtain JARs via [Maven Central](https://search.maven.org/search?q=g:org.scala-lang%20AND%20v:2.13.11)
```
Comment (xuwei-k) [inline releases/2.13.12.md]:
```suggestion
* Download a distribution from [scala-lang.org](https://scala-lang.org/download/2.13.12.html)
* Obtain JARs via [Maven Central](https://search.maven.org/search?q=g:org.scala-lang%20AND%20v:2.13.12)
```

**Comment by som-snytt** [comment]
As there is no nit too picky, the git query does not account for co-authorship, which I believe counts for unit contribution and not `0.5`. This more correct accounting vaults Georgi in front of Nicolas, in alphabetical suborder.

Lukas also shares a co-authorship credit, but it seems a couple of his PRs are sponsored, so I'm not sure how that adds up. I think it means Lukas takes every one out for biers?

File: releases/2.13.12.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+## Scala 2.13 notes
+
+The [release notes for Scala 2.13.0](https://github.com/scala/scala/releases/v2.13.0) have important information applicable to the whole 2.13 series.
+
+## Obtaining Scala
+
+Scala releases are available through a variety of channels, including (but not limited to):
+
+* Bump the `using scala` setting in your Scala CLI project
+* Bump the `scalaVersion` setting in your sbt project
```
Comment (lefou) [inline releases/2.13.12.md]: Let's also guide all those Mill users out there.

```suggestion
* Bump the `scalaVersion` setting in your sbt project
* Bump the `scalaVersion` setting in your Mill project
```

Comment (som-snytt) [reply releases/2.13.12.md]: "... in the build tool of your choice ..." or "... respectively, what they use at the office."

Comment (SethTisue) [reply releases/2.13.12.md]: I went with "sbt or Mill". We do have the "(but not limited to):" disclaimer.

**Comment by SethTisue** [comment]
> the git query does not account for co-authorship

@som-snytt hmm... yeah, that's not ideal. I would use a more correct automated method if one existed, but I don't think I'm sufficiently motivated to build one myself.

**Comment by SethTisue** [comment]
this is now live at https://github.com/scala/scala/releases/tag/v2.13.12

It's not too late to make further edits — feel free to continue commenting here.

**Comment by som-snytt** [comment]
per discussion on discord chat, perhaps the item on `scala3-migration` noise deserves a note that changes in behavior were made noisier, and that it's expected to `-Wconf:cat=scala3-migration:s` in some form.

My first comment was that for override inference, Scala 3 style is desirable, but is not backward-compatible. The user should be aware of the difference; the desirable action is to make the old inferred type explicit (for compatibility) or adjust the warning, if the new behavior is acceptable.
```
scala 2.13.12> trait T { def f: Option[Int] }
trait T

scala 2.13.12> class C extends T { def f = None }
                                       ^
               error: under -Xsource:3, inferred Option[Int] instead of None.type [quickfixable]
               Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
               Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=C.f

scala 2.13.12> :replay -Wconf:cat=scala3-migration:w
replay> trait T { def f: Option[Int] }
trait T


scala 2.13.12> class C extends T { def f = None }
                                       ^
               warning: under -Xsource:3, inferred Option[Int] instead of None.type [quickfixable]
class C
```
The other example in chat was the access modifiers on case class methods when the constructor is private.
```
scala 2.13.12> case class C private (c: String)
                          ^
               error: constructor modifiers are assumed by synthetic `copy` method
               Scala 3 migration messages are errors under -Xsource:3. Use -Wconf / @nowarn to filter them or add -Xmigration to demote them to warnings.
               Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=scala3-migration, site=C

scala 2.13.12> :replay -Wconf:cat=scala3-migration:w
Nothing to replay.

scala 2.13.12> case class C private (c: String)
                          ^
               warning: constructor modifiers are assumed by synthetic `copy` method
                          ^
               warning: constructor modifiers are assumed by synthetic `apply` method
class C
```
If it's desirable to keep the old access, it would be necessary to write the methods by hand.

It would be nice to have these use cases in the release notes.

Example `-Xsource:3` which does not make noise:
```
scala 2.13.12> import collection.*
import collection._
```

**Comment by SethTisue** [comment]
@som-snytt maybe you and I could work together (not this week) on documenting `-Xsource:3` properly. maybe with three lists: enabled behaviors, modified behaviors, disabled/discouraged/deprecated behaviors... something like that. and a good introductory text that explains what the flag is and why, and include any useful usage advice about `-Xmigration` and `-Wconf` and so on.

In the meantime, though, if you have any suggested edits to the release notes themselves, feel free to go ahead and make them. but let's not add too much. or if we do find we've added too much, let's have a goal of moving it to a more permanent location.

**Comment by SethTisue** [comment]
@som-snytt oh, I guess the broader topic of documenting `-Xsource:3` should continue on https://github.com/scala/scala-dev/issues/470

**Comment by SethTisue** [comment]
@som-snytt the release notes now have:

* Emit migration warnings under `-Xsource:3` as fatal warnings, not errors ([#10439](https://github.com/scala/scala/pull/10439) by [@som-snytt](https://github.com/som-snytt), [#10511](https://github.com/scala/scala/pull/10511))
  * `-Xmigration` will demote the errors to warnings
  * `-Wconf:cat=scala3-migration:s` will silence these errors/warnings altogether

I remain interested in properly documenting `-Xsource:3` (but not this week or next).

**Pull request closed without merging**